granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -78,7 +95,99 @@ def find_v2_hierarchy(resolve_host_root_links: bool = True) -> Optional[str]:
     return path
 
 
-def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> Optional[str]:
+CGROUP_PROCS_FILE = "cgroup.procs"
+
+
+class CgroupCore:
+    path: Path
+
+    def __init__(self, path: Path):
+        self.path = path
+
+    def _get_parent_cgroup(self, cgroup_name: str) -> Optional[Path]:
+        cgroup_index = self.path.parts.index(cgroup_name)
+        if cgroup_index != -1:
+            return Path(*self.path.parts[: cgroup_index + 1])
+        return None
+
+    def _create_subcgroup(self, cgroup_name: str) -> Optional[Path]:
+        if self.has_parent_cgroup(cgroup_name):
+            new_cgroup_path = self._get_parent_cgroup(cgroup_name)
+        else:
+            new_cgroup_path = self.path / cgroup_name
+            new_cgroup_path.mkdir(exist_ok=True)
+
+        return new_cgroup_path
+
+    def has_parent_cgroup(self, cgroup_name: str) -> bool:
```

Comment:
I'd avoid re-implementing the "parts" logic and implement this function as `return _get_parent_cgroup(name) is not None`